rust/kernel/io_mem.rs

```diff
@@ -177,6 +177,47 @@ impl<const SIZE: usize> IoMem<SIZE> {
         crate::build_assert!(Self::offset_ok::<T>(offset), "IoMem offset overflow");
     }
 
+    /// Copy memory block from an i/o memory into a buffer.
+    ///
+    /// # Examples
+    /// ```
+    /// # use kernel::prelude::*;
+    /// use kernel::io_mem::{self, IoMem, Resource};
+    ///
+    /// fn test(res: Resource) -> Result {
+    ///     // Create an i/o memory block of at least 100 bytes.
+    ///     let mem = unsafe { IoMem::<100>::try_new(res) }?;
+    ///
+    ///     let mut buffer: [i8; 32] = [0; 32];
+    ///
+    ///     // Memcpy 16 bytes from offset 10 of i/o memory block into the buffer.
+    ///     mem.try_memcpy_fromio(&mut buffer, 10, 16)?;
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn try_memcpy_fromio(&self, from: &mut [i8], offset: usize, count: usize) -> Result {
```

Comment:
Instead of taking an explicit `count` argument, why not use the length of the buffer?

Callers that have a larger buffer can index it with a range to "shrink" it to the size they want, for example:
```rust
mem.try_memcpy_fromio(&mut buff[..count], offset)?;
```

I think this is less error prone because the caller will have fewer opportunities to confuse arguments based on order, for example when reading or writing `mem.try_memcpy_fromio(v, 10, 20)`, one might wonder: is 10 the offset and 20 the count or vice-versa?

Reply:
Slicing looks very expressive here. This causes a small change in semantics but it will be documented with an example.